clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -1468,6 +1467,145 @@ test_replay_skip()
 #endif
 }
 
+static void
+test_replay_as_traced()
+{
+#ifdef HAS_ZIP
+    std::cerr << "\n----------------\nTesting replay as-traced\n";
+
+    static constexpr int NUM_INPUTS = 4;
+    static constexpr int NUM_OUTPUTS = 2;
+    static constexpr int NUM_INSTRS = 9;
+    static constexpr memref_tid_t TID_BASE = 100;
+    static constexpr int CPU0 = 6;
+    static constexpr int CPU1 = 9;
+    std::vector<trace_entry_t> inputs[NUM_INPUTS];
+    for (int i = 0; i < NUM_INPUTS; i++) {
+        memref_tid_t tid = TID_BASE + i;
+        inputs[i].push_back(make_thread(tid));
+        inputs[i].push_back(make_pid(1));
+        for (int j = 0; j < NUM_INSTRS; j++)
+            inputs[i].push_back(make_instr(42 + j * 4));
+        inputs[i].push_back(make_exit(tid));
+    }
+
+    // Synthesize a cpu-schedule file.
+    std::string cpu_fname = "tmp_test_cpu_as_traced.zip";
+    static const char *const CORE0_SCHED_STRING = "AAACCCAAACCCBBBDDD";
```

Comment:
Do we verify in any way if the BBB in core0 are the instrs 7-9 in B's instrs stream? Could they be instrs 4-6?

Reply:
Oh it's executes lockstep, so the fact that the index of Bs in core0 is > index of Bs in core1, that ensures it.

Reply:
Right.